Can't use inside c++ code

gcc failed to compile:
```shell
/home/ksergey/dev/BithumbGateway/deps/fastbase64/include/chromiumbase64.h: In function ‘std::__cxx11::string& chromium_base64_encode(std::__cxx11::string&)’:
/home/ksergey/dev/BithumbGateway/deps/fastbase64/include/chromiumbase64.h:162:21: error: redefinition of ‘std::__cxx11::string& chromium_base64_encode(std::__cxx11::string&)’
 inline std::string& chromium_base64_encode(std::string& s)
                     ^
/home/ksergey/dev/BithumbGateway/deps/fastbase64/include/chromiumbase64.h:144:21: note: ‘std::__cxx11::string& chromium_base64_encode(std::__cxx11::string&)’ previously defined here
 inline std::string& chromium_base64_encode(std::string& s)

```

```c++
...
#ifdef __cplusplus                                                                                                                                             
}                                                                                                                                                              
                                                                                                                                                               
#include <string>                                                                                                                                              
                                                                                                                                                               
inline std::string& chromium_base64_encode(std::string& s)                                                                                                     
{                                                                                                                                                              
    std::string x(chromium_base64_encode_len(s.size()), '\0');                                                                                                 
    size_t d = chromium_base64_encode(const_cast<char*>(x.data()), s.data(), (int)s.size());                                                                       x.erase(d, std::string::npos);                                                                                                                             
    s.swap(x);                                                                                                                                                 
    return s;                                                                                                                                                  
}                                                                                                                                                                                                                                                                                                                             
/**                                                                                                                                                            
 * base 64 decode a string (self-modifing)                                                                                                                     
 * On failure, the string is empty.                                                                                                                            
 *                                                                                                                                                             
 * This function is for C++ only (duh)                                                                                                                         
 *                                                                                                                                                             
 * \param[in,out] s the string to be decoded                                                                                                                   
 * \return a reference to the input string                                                                                                                     
 */                                                                                                                                                            
inline std::string& chromium_base64_encode(std::string& s)                                                                                                     
{                                                                                                                                                              
    std::string x(chromium_base64_encode_len(s.size()), '\0');                                                                                                 
    size_t d = chromium_base64_encode(const_cast<char*>(x.data()), s.data(), (int)s.size());                                                                   
    if (d == MODP_B64_ERROR) {                                                                                                                                 
        x.clear();                                                                                                                                             
    } else {                                                                                                                                                   
        x.erase(d, std::string::npos);                                                                                                                         
    }                                                                                                                                                          
    s.swap(x);                                                                                                                                                 
    return s;                                                                                                                                                  
}                                                                                                                                                              
                                                                                                                                                               
#endif /* __cplusplus */                                                                                                                                       
#endif                                                                                                                                                        
...
```

--- a/include/fastavxbase64.h
+++ b/include/fastavxbase64.h
@@ -9,6 +9,10 @@
 #include <stdint.h>
 #include "chromiumbase64.h"
 
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
+
 /**
 * This code extends Nick Galbreath's high performance base 64decoder (used in Chromium), the API is the
 * same effectively, see chromium64.h.
@@ -26,6 +30,8 @@ size_t fast_avx2_base64_decode(char *out, const char *src, size_t srclen);
 */
 size_t fast_avx2_base64_encode(char* dest, const char* str, size_t len);
 
-
+#ifdef __cplusplus
+}
+#endif /* __cplusplus */
 
 #endif

--- a/include/klompavxbase64.h
+++ b/include/klompavxbase64.h
@@ -8,6 +8,9 @@
 #include <stddef.h>
 #include <stdint.h>
 
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
 
 /**
 * This code is based on Alfred Klomp's https://github.com/aklomp/base64 (published under BSD)
@@ -66,5 +69,8 @@ void klomp_avx2_base64_encode
 	, size_t		*outlen
 	) ;
 
+#ifdef __cplusplus
+}
+#endif /* __cplusplus */
 
 #endif

--- a/include/scalarbase64.h
+++ b/include/scalarbase64.h
@@ -8,6 +8,9 @@
 #include <stddef.h>
 #include <stdint.h>
 
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
 
 /**
 * We copy the code below from https://github.com/aklomp/base64
@@ -37,5 +40,8 @@ void scalar_base64_encode
 	, size_t		*outlen
 	) ;
 
+#ifdef __cplusplus
+}
+#endif /* __cplusplus */
 
 #endif

